app/server/app/routes/api.js

```diff
@@ -32,19 +33,47 @@ router.get("/bap", (req, res, next) => {
   ]);
 });
 
-router.get("/form-schema", (req, res, next) => {
-  // throw new Error("TODO: implement Forms.gov integration");
-  res.json({ schema: "TODO" });
+router.get("/form-schema", ensureAuthenticated, (req, res) => {
+  // If "version" is passed in querystring, add to form.io request to get specific version's schema
+  let url = process.env.FORMIO_BASE_URL;
+  if (req.query.version) {
+    url += `/v/${req.query.version}`;
+  }
+  axios
+    .get(url, {
+      headers: {
+        "x-token": process.env.FORMIO_API_KEY,
+      },
+    })
+    .then((response) => {
+      res.json(response.data);
```

Comment:
FYI, the forms schema/definition we need would be the entire JSON object returned in the response (there's no `data` key returned in this request – that's just for the submissions)

Reply:
The response.data part is an axios thing - the axios response is an object that includes the request, config, the actual response data, and some other info (https://axios-http.com/docs/res_schema)

Reply:
oh wow, thank you...shows how much I use axios. Coincidentally, a Form.io submission stores the submitted form fields data in a `data` object (along with other form metadata alongside it), so I assumed that's why you used that.

That said, the submissions returned are an array of objects too – so just ignore what I said, and plan to pass the response along as-is just like you're doing, and I can parse it as needed from the client app :smile: